osu.Game/Overlays/RestoreDefaultValueButton.cs

```diff
@@ -20,15 +20,26 @@ public class RestoreDefaultValueButton<T> : OsuButton, IHasTooltip, IHasCurrentV
     {
         public override bool IsPresent => base.IsPresent || Scheduler.HasPendingTasks;
 
-        private readonly BindableWithCurrent<T> current = new BindableWithCurrent<T>();
-
         // this is done to ensure a click on this button doesn't trigger focus on a parent element which contains the button.
         public override bool AcceptsFocus => true;
 
+        // this is intentionally not using BindableWithCurrent, as it can use the wrong IsDefault implementation when passed a BindableNumber.
+        // using GetBoundCopy() ensures that the received bindable is of the exact same type as the source bindable and uses the proper IsDefault implementation.
+        private Bindable<T> current;
+
         public Bindable<T> Current
         {
-            get => current.Current;
-            set => current.Current = value;
+            get => current;
+            set
+            {
+                current?.UnbindAll();
+                current = value.GetBoundCopy();
+
+                current.ValueChanged += _ => UpdateState();
+                current.DefaultChanged += _ => UpdateState();
+                current.DisabledChanged += _ => UpdateState();
+                UpdateState();
```

Comment:
Think this might need a `IsLoaded` guard still, and the `LoadComplete()` stuff might need to go back in? Not sure.

Reply:
It's sorta unnecessary since it's already injected with a `Current` once created in `SettingsItem`, but no point in removing it.